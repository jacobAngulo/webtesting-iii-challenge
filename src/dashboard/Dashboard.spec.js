// Test away!
import React from "react";
import { render, fireEvent, cleanup } from "react-testing-library";
import renderer from "react-test-renderer";
import "react-testing-library/cleanup-after-each";
import Dashboard from "./Dashboard";
import Display from "../display/Display";
import 'jest-dom/extend-expect'

describe("<Dashboard />", () => {
  it("matches snapshot", () => {
    const tree = renderer.create(<Dashboard />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("is mocking me", () => {
    const mock = jest.fn();

    const result = mock();

    expect(result).toBeUndefined();
    expect(mock).toHaveBeenCalledTimes(1);
  });

  it("should initially be unlocked and open", () => {
    const { getByTestId } = render(<Display />);
    const lock = getByTestId("lock-display");
    const gate = getByTestId("close-display");

    expect(lock).toHaveTextContent("Unlocked");
    expect(lock).toHaveClass("green-led");
    expect(gate).toHaveTextContent("Open");
    expect(gate).toHaveClass("green-led");
  });

  it("Should lock and unlock as expected", () => {
    const { getByTestId } = render(<Dashboard />);
    const lock = getByTestId("lock-display");
    const gate = getByTestId("close-display");

    fireEvent.click(getByTestId("close-btn"));
    expect(lock).toHaveTextContent("Unlocked");
    expect(gate).toHaveTextContent("Closed");
    expect(lock).toHaveClass("green-led");
    expect(gate).toHaveClass("red-led");

    fireEvent.click(getByTestId("lock-btn"));
    expect(lock).toHaveTextContent("Locked");
    expect(gate).toHaveTextContent("Closed");
    expect(lock).toHaveClass("red-led");
    expect(gate).toHaveClass("red-led");
  });
});
