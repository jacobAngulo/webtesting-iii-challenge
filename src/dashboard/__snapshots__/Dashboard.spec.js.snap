// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dashboard /> matches snapshot 1`] = `
Array [
  <div
    className="display panel"
  >
    <div
      className="led green-led"
      data-testid="lock-display"
    >
      Unlocked
    </div>
    <div
      className="led green-led"
      data-testid="close-display"
    >
      Open
    </div>
  </div>,
  <div
    className="controls panel"
  >
    <button
      className="toggle-btn"
      data-testid="lock-btn"
      disabled={true}
      onClick={[Function]}
    >
      Lock Gate
    </button>
    <button
      className="toggle-btn"
      data-testid="close-btn"
      disabled={false}
      onClick={[Function]}
    >
      Close Gate
    </button>
  </div>,
]
`;
